Lib/ftplib.py

```diff
@@ -324,7 +324,10 @@ def makeport(self):
     def makepasv(self):
         """Internal: Does the PASV or EPSV handshake -> (address, port)"""
         if self.af == socket.AF_INET:
-            untrusted_host, port = parse227(self.sendcmd('PASV'))
+            try:
+                untrusted_host, port = parse227(self.sendcmd('PASV'))
+            except error_reply:
+                untrusted_host, port = parse229(self.sendcmd('EPSV'), self.sock.getpeername())
```

Comment:
Sorry for the back-and-forth, but since GP accepted the previous alternative, let's send 'PASV' normally and accept both 227 and 229 response codes instead.

Reply:
Sure. Changed as GP suggested.